File: 4_SEMESTRE/SI400_Programacao_Orientada_Objetos_II/manipulacao_entidades_geometricas/Coordenada.java
public class Coordenada {
    private int x;
    private int y;

    public Coordenada(int x, int y) {
        this.x = x;
        this.y = y;
    }

    @Override
    public String toString() {
        return "(" + this.x + ", " + this.y + ")";
    }

    public static double distanciaCoordenadas(Coordenada c1, Coordenada c2) {
        double deltaX = c1.x - c2.x;
        double deltaY = c1.y - c2.y;
        double cateto1 = Math.pow(deltaX, 2);
        double cateto2 = Math.pow(deltaY, 2);
        return Math.sqrt(cateto1 + cateto2);
    }
    
    public void translacaoCoordenada(int deltaX, int deltaY) {
        this.x += deltaX;
        this.y += deltaY;
    }
}

File: 4_SEMESTRE/TT002_Python/Tarefa08_09_10/Coordenada.py
import math

class Coordenada:
    def __init__(self, *args):
        if len(args) > 1:
            raise ValueError(f"Numero de argumentos errado: {len(args)}")
        coordenada = args[0]
        if not isinstance(coordenada, tuple):
            raise TypeError("Parametro nao eh uma tupla")
        if len(coordenada) != 2:
            raise ValueError("A tupla deve possuir dois elementos e os elementos devem ser numeros")
        if not all(isinstance(c, (int, float)) for c in coordenada):
            raise TypeError("Os elementos da tupla devem ser inteiros ou flutuantes. ")
        self.x, self.y = coordenada

    def distancia(self, coordenada):
        deltax = self.x - coordenada.x
        deltay = self.y - coordenada.y
        distancia = math.sqrt((deltax**2 + deltay**2))
        return distancia

    def __str__(self):
        return f"({self.x},{self.y})"